pkg/utils/imagefilterlist.go

```diff
@@ -0,0 +1,142 @@
+package utils
+
+import (
+	"errors"
+	"github.com/gobwas/glob"
+	"log"
+	"regexp"
+	"strings"
+)
+
+const defaultContainerRegistry = "docker.io"
+
+// ImageFilterList represents a list of glob filters that can be used to check if an image matches one of these filters
+type ImageFilterList struct {
+	registryMatcher *regexp.Regexp
+	userMatcher     *regexp.Regexp
```

Comment:
I think we can simplify the checks without having specific regex for registry and user (which may or may not be present in the image name for example `alpine`, `ubuntu`, `mongo` etc. ).

I would propose a simpler algorithm like the following:

- check if the image name matches the allow list, if true return without further processing
- if the image does not match, try splitting on the first `/` and check if the first token is a hostname (optionally removing the port) using https://pkg.go.dev/net#LookupHost or https://pkg.go.dev/net#Resolver.LookupHost (depending if we need to specify a context or if we are ok with context.Background)
- if the first token is not a hostname we can try prepending the `docker.io` default for the docker public image registry and try matching against the allow list again (optionally we could try a HEAD http request against the docker registry but it's more expensive and we may incur in docker registry rate limit)

Reply:
I think checking the hostname and verifying that the image is present at the docker registry is a little bit out of scope, but I look into simplifying the matching of the images